internal/model/user_server.go

```diff
@@ -160,10 +160,22 @@ func (this *ServerUser) GetCourseRole(courseID string) CourseUserRole {
 }
 
 // Convert this server user into a course user for the specific course.
+// Set courseOnly to false if high level server users should be converted to a course user.
 // Will return (nil, nil) if the user is not enrolled in the given course.
-func (this *ServerUser) ToCourseUser(courseID string) (*CourseUser, error) {
+func (this *ServerUser) ToCourseUser(courseID string, courseOnly bool) (*CourseUser, error) {
 	info, exists := this.CourseInfo[courseID]
 	if !exists {
+		if !courseOnly && this.Role >= ServerRoleAdmin {
```

Comment:
I'm not a fan of the current conditional structure,
it has more corners than I would like.

Instead you can store whether or not to escalate in a variable.
Then you can have something like:
```
if (!enrolled && !escalate) {
    return nil, nil;
}

courseUser := ... basic info ...

if (enrolled) {
    // Set course info.
}

if (escalate) {
    // Set role.
}

return courseUser, nil;